ogre/include/ignition/rendering/ogre/OgreHeightmap.hh

```diff
@@ -0,0 +1,126 @@
+    // Forward declaration
+    class OgreHeightmapPrivate;
+
+    /// \brief Ogre implementation of a heightmap geometry.
+    class IGNITION_RENDERING_OGRE_VISIBLE OgreHeightmap
+      : public BaseHeightmap<OgreGeometry>
+    {
+      /// \brief Constructor
+      protected: explicit OgreHeightmap(const HeightmapDescriptor &_desc);
+
+      /// \brief Destructor
+      public: virtual ~OgreHeightmap();
+
+      // Documentation inherited.
+      public: virtual void Init() override;
+
+      // Documentation inherited.
+      public: virtual void PreRender() override;
+
+      /// \brief Returns NULL, heightmaps don't have movable objects.
+      /// \return Null pointer.
+      public: virtual Ogre::MovableObject *OgreObject() const override;
+
+      /// \brief Returns NULL, heightmap materials don't inherit from
+      /// MaterialPtr.
+      /// \return Null pointer.
+      public: virtual MaterialPtr Material() const override;
+
+      /// \brief Has no effect for heightmaps. The material is set through a
+      /// HeightmapDescriptor.
+      /// \param[in] _material Not used.
+      /// \param[in] _unique Not used.
+      public: virtual void SetMaterial(MaterialPtr _material, bool _unique)
+          override;
+
+      /// \brief Configure the terrain default values.
+      /// \todo Move to private?
+      private: void ConfigureTerrainDefaults();
+
+      /// \brief Checks if the terrain was previously loaded by comparing its
+      /// hash against the one stored in the terrain directory
+      /// \param[in] _terrainDirPath Path to the directory containing the
+      /// terrain files and hash.
+      /// \return True if the terrain requires to regenerate the terrain files.
+      private: bool PrepareTerrain(const std::string &_terrainDirPath);
+
+      /// \brief Update the hash of a terrain file. The hash will be written in
+      /// a file called gzterrain.SHA1 . This method will be used when the
+      /// paging is enabled and the terrain is loaded for the first time or if
+      /// the heightmap's image has been modified.
+      /// \param[in] _hash New hash value
+      /// \param[in] _terrainDir Directory where the terrain hash and the
+      /// terrain pages are stored. Ex: $TMP/gazebo-paging/heigthmap_bowl
+      private: void UpdateTerrainHash(const std::string &_hash,
+          const std::string &_terrainDir);
+
+      /// \brief Split a terrain into subterrains
+      /// \param[in] _heightmap Source vector of floats with the heights.
+      /// \param[in] _n Number of subterrains.
+      /// \param[out] _v Destination vector with the subterrains.
+      private: void SplitHeights(const std::vector<float> &_heightmap,
+                  const int _n, std::vector<std::vector<float>> &_v);
+
+      /// \brief Define a section of the terrain.
+      /// \param[in] _x X coordinate of the terrain.
+      /// \param[in] _y Y coordinate of the terrain.
+      private: void DefineTerrain(const int _x, const int _y);
```

Comment:
I think we can leave out `const` for primitive types. I've started doing that to be consistent with other libraries.